ui/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/ManagementActor.scala

```diff
@@ -120,11 +120,11 @@ class ManagementActor(environment: String,
 
   //This method handles some corner cases like retention for keeping old states - some engine can cleanup canceled states. It's more Flink hermetic.
   //TODO: In future we should move this functionality to ProcessManager.
-  private def handleObsoleteStatus(processState: Option[ProcessState], processStateDefinitionManager: ProcessStateDefinitionManager, lastAction: Option[ProcessDeploymentAction]): Option[ProcessStatus] =
+  private def handleObsoleteStatus(processStateDefinitionManager: ProcessStateDefinitionManager, processState: Option[ProcessState], lastAction: Option[ProcessDeploymentAction]): ProcessStatus =
     (processState, lastAction) match {
-      case (Some(state), _)  => Option(ProcessStatus.create(state, lastAction))
-      case (None, Some(action)) if action.isCanceled => Option(ProcessStatus.canceled(processStateDefinitionManager))
-      case _ => Option.empty
+      case (Some(state), _)  => ProcessStatus.create(state, lastAction)
+      case (None, Some(action)) if action.isCanceled => ProcessStatus(SimpleStateStatus.Canceled, processStateDefinitionManager)
+      case _ => ProcessStatus(SimpleStateStatus.NotDeployed, processStateDefinitionManager)
```

Comment:
please explicitly handle `(None, None)`, there are some cases that we are mathing now as a not deployed but it isn't "perfect match" ;-)

Reply:
You'r right, changed! ;)